src/SecTester.Scan/CI/VendorMatcher.cs

```diff
@@ -0,0 +1,89 @@
+using System.Collections;
+using System.Linq;
+using System.Text.Json;
+
+namespace SecTester.Scan.CI;
+
+internal class VendorMatcher
+{
+  private readonly IDictionary _env;
+
+  public VendorMatcher(IDictionary env)
+  {
+    _env = env;
+  }
+
+  public bool MatchEnv(JsonElement element)
+  {
+    return element.ValueKind switch
+    {
+      JsonValueKind.String => CheckStringValue(element),
+      JsonValueKind.Object => Any(element) || Includes(element) || HasOwnProperties(element),
+      JsonValueKind.Array => element.EnumerateArray().All(CheckStringValue),
+      _ => false
+    };
+  }
+
+  public bool MatchPr(JsonElement element)
+  {
+    return element.ValueKind switch
+    {
+      JsonValueKind.String => CheckStringValue(element),
+      JsonValueKind.Object => NotEqual(element) || MatchEnv(element),
```

Comment:
What is the point to call the whole `MatchEnv`? Extract `JsonValueKind.Object => Any(element) || Includes(element) || HasOwnProperties(element)` to a separate method